packages/react/src/lib/accessibility/Behaviors/Dialog/dialogBehavior.ts

```diff
@@ -15,8 +14,6 @@ import * as _ from 'lodash'
  * Adds attribute 'aria-labelledby' based on the property 'aria-labelledby' to 'popup' component's part.
  * Adds attribute 'aria-describedby' based on the property 'aria-describedby' to 'popup' component's part.
  * Adds attribute 'role=dialog' to 'popup' component's part.
- * Generates unique ID and adds it as attribute 'id' to the 'header' component's part if it has not been provided by the user.
- * Generates unique ID and adds it as attribute 'id' to the 'content' component's part if it has not been provided by the user.
```

Comment:
This definitely should be replaced. @silviuavram @sophieH29 can you help me with rules?

Reply:
Maybe you can refactor the `Dialog-test.tsx` to check that IDs are generated there if they are not provided. I think you can remove the other tests (label, labelledby etc) since they should be checked in the behavior.

In `dialogBehavior-test.tsx` you can add individual tests for the generation. Pass `header` as a shorthand with `id` and check it's added as `aria-labelledby`. Similar to `content`. The other case when they are applied directly from the `aria-labelledby` / `aria-describedby` should already be tested by a rule.

After looking at this file at `'aria-labelledby': defaultAriaLabelledBy || props['aria-labelledby'],` probably you can remove the `|| props['aria-labelledby']` and do the whole logic in the method, which you can rename without `Default`: `getAriaDescribedBy` or something. Where you return it as the prop value, if passed, as header['id] if header exists, or nothing.

Reply:
what do you think @sophieH29 ?

Reply:
Updated UTs 🔍